File: Level-0/65.java
import java.util.*;
class Solution {
    public int[] solution(int[] numlist, int n) {
        int[] answer = new int[numlist.length];
        HashMap<Integer, Integer> map = new HashMap<Integer, Integer>();
        for(int i = 0; i < numlist.length; i++){
            map.put(numlist[i], Math.abs(n - numlist[i]));
        }
        
        ArrayList<Integer> key = new ArrayList<Integer>(map.keySet());
        Collections.sort(key, Collections.reverseOrder());
        key.sort((a, b) -> Math.abs(n - a) - Math.abs(n - b));
        
        for(int i = 0; i < key.size(); i++){
            answer[i] = key.get(i);
        }
        
        return answer;
    }
}

File: Level-1/65.js
function solution(k, score) {
    const answer = [];
    const rank = [];
    
    score.map((v, i) => {
        if(i < k) rank.push(v);
        else {
            const min = rank[rank.length - 1];
            if(min < v) {
                rank.pop();
                rank.push(v);
            }
        }
        rank.sort((a, b) => b - a);
        answer.push(rank[rank.length - 1]);
    });
    
    return answer;
}
